framework/src/actions/CheckOutputAction.C

```diff
@@ -50,10 +50,10 @@ CheckOutputAction::checkVariableOutput(const std::string & task)
   {
     // Loop through the actions for the given task
     const std::vector<Action *> & actions = _awh.getActionsByName(task);
-    for (std::vector<Action *>::const_iterator it = actions.begin(); it != actions.end(); ++it)
+    for (auto act : actions)
```

Comment:
I've basically decided to use `auto &` for loops where I'm going to change the thing I'm iterating over and `const auto &` where I'm not.

The other option is `auto &&` and `const auto &&` which is more general...